app/controllers/concerns/course/lesson_plan/strategies/fixed_personalization_strategy.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+class FixedPersonalizationStrategy < BasePersonalizationStrategy
+  # Returns a hash containing lesson plan item ids to submission time.
+  #
+  # @param [CourseUser] course_user The course user to compute data for.
+  # @return [Hash{Integer=>ActiveSupport::TimeWithZone|nil}] A hash of submitted lesson plan items' IDs to their
+  #   submitted time, if relevant/available.
+  def precompute_data(course_user)
+    lesson_plan_items_submission_time_hash(course_user)
+  end
+
+  # Deletes all personal times that are not fixed or submitted. This basically causes the course user to follow the
+  # reference timeline moving forward.
+  #
+  # @param [CourseUser] course_user The course user to compute data for.
+  # @param [Hash{Integer=>ActiveSupport::TimeWithZone|nil}] precompute_data A hash of submitted lesson plan items' ID to
```

Comment:
I think the input type can be standardised across different classes to avoid confusion..

Reply:
Personally I think it's fine, so long every strategy generates its own precomputed data and is able to execute on that. We definitely can standardise if we want to, but I feel that allowing the strategy to decide their own required data keeps the strategies flexible.